tutorials/plot_background_filtering.py

```diff
@@ -0,0 +1,365 @@
+morlet_freq = f_p
+tlim = [center - 0.2, center + 0.2]
+tticks = [tlim[0], center, tlim[1]]
+flim = [20, 70]
+
+x = np.zeros(int(sfreq * dur))
+blip = morlet(sfreq, [morlet_freq], n_cycles=7)[0].imag / 20.
+n_onset = int(center * sfreq) - len(blip) // 2
+x[n_onset:n_onset + len(blip)] += blip
+x_orig = x.copy()
+
+rng = np.random.RandomState(0)
+x += rng.randn(len(x)) / 1000.
+x += np.sin(2. * np.pi * 60. * np.arange(len(x)) / sfreq) / 2000.
+
+###############################################################################
+# Filter it with a shallow cutoff, linear-phase FIR and compensate for
+# the delay:
+
+transition_band = 0.25 * f_p
+f_s = f_p + transition_band
+filter_dur = 5. / transition_band  # sec
+n = int(sfreq * filter_dur)
+freq = [0., f_p, f_s, sfreq / 2.]
+gain = [1., 1., 0., 0.]
+h = signal.firwin2(n, freq, gain, nyq=sfreq / 2.)
+x_shallow = np.convolve(h, x)[len(h) // 2:]
+
+###############################################################################
+# Now let's filter it with the MNE-Python 0.12 defaults, which is a
+# long-duration, steep cutoff FIR:
+
+transition_band = 0.5  # Hz
+f_s = f_p + transition_band
+filter_dur = 10.  # sec
+n = int(sfreq * filter_dur)
+freq = [0., f_p, f_s, sfreq / 2.]
+gain = [1., 1., 0., 0.]
+h = signal.firwin2(n, freq, gain, nyq=sfreq / 2.)
+x_steep = np.convolve(h, x)[len(h) // 2:]
+
+plot_filter(h, 'MNE-Python 0.12 default', freq, gain)
+
+###############################################################################
+# It has excellent frequency attenuation, but this comes at a cost of potential
+# ringing (long-lasting ripples) in the time domain. Ripple can occur with
+# steep filters, especially on signals with frequency content around the
+# transition band. Our Morlet wavelet signal has power in our transition band,
+# and the time-domain ringing is thus more pronounced for the steep-slope,
+# long-duration filter than the shorter, shallower-slope filter:
+
+axs = plt.subplots(2)[1]
+
+
+def plot_signal(x, offset):
+    t = np.arange(len(x)) / sfreq
+    axs[0].plot(t, x + offset)
+    axs[0].set(xlabel='Time (sec)', xlim=t[[0, -1]])
+    box_off(axs[0])
+    X = fftpack.fft(x)
+    freqs = fftpack.fftfreq(len(x), 1. / sfreq)
+    mask = freqs >= 0
+    X = X[mask]
+    freqs = freqs[mask]
+    axs[1].plot(freqs, 20 * np.log10(np.abs(X)))
+    axs[1].set(xlim=xlim)
+
+yticks = np.arange(4) / -30.
+yticklabels = ['Original', 'Noisy', 'FIR-shallow$', 'FIR-steep$']
+plot_signal(x_orig, offset=yticks[0])
+plot_signal(x, offset=yticks[1])
+plot_signal(x_shallow, offset=yticks[2])
+plot_signal(x_steep, offset=yticks[3])
+axs[0].set(xlim=tlim, title='Lowpass=%d Hz' % f_p, xticks=tticks,
+           ylim=[-0.125, 0.025], yticks=yticks, yticklabels=yticklabels,)
+for text in axs[0].get_yticklabels():
+    text.set(rotation=45, size=8)
+axs[1].set(xlim=flim, ylim=ylim, xlabel='Frequency (Hz)',
+           ylabel='Magnitude (dB)')
+box_off(axs[0])
+box_off(axs[1])
+mne.viz.tight_layout()
+
```

Comment:
Intuitively, I'd've expected some sort of conclusion here. Like, "To sum it up, ars longa, vita brevis, filtering is hard, you should have gotten a real job instead. Have fun with your data."

Or alternatively something that's not insulting, but instead helpful.